src/devices/Sht3x/Sht3x.cs

```diff
@@ -0,0 +1,169 @@
+        /// </summary>
+        public bool Heater
+        {
+            get => _heater;
+            set
+            {
+                SetHeater(value);
+                _heater = value;
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the SHT3x
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        /// <param name="resolution">SHT3x Read Resolution</param>
+        public Sht3x(I2cDevice sensor, Resolution resolution = Resolution.High)
+        {
+            _sensor = sensor;
+
+            Resolution = resolution;
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            _sensor?.Dispose();
+            _sensor = null;
+        }
+
+        /// <summary>
+        /// SHT3x Soft Reset
+        /// </summary>
+        public void Reset()
+        {
+            Write(Register.SHT_RESET);
+        }
+
+        /// <summary>
+        /// Set SHT3x Heater
+        /// </summary>
+        /// <param name="isOn">Heater on when value is true</param>
+        private void SetHeater(bool isOn)
+        {
+            if (isOn)
+                Write(Register.SHT_HEATER_ENABLE);
+            else
+                Write(Register.SHT_HEATER_DISABLE);
+        }
+
+        /// <summary>
+        /// Read Temperature and Humidity
+        /// </summary>
+        private void ReadTempAndHumi()
+        {
+            Span<byte> writeBuff = stackalloc byte[] { (byte)Register.SHT_MEAS, (byte)Resolution };
+            Span<byte> readBuff = stackalloc byte[6];
+
+            _sensor.Write(writeBuff);
+            Thread.Sleep(20);       // wait SCL free
+            _sensor.Read(readBuff);
+
+            // Details in the Datasheet P13
+            int st = (readBuff[0] << 8) | readBuff[1];      // Temp
+            int srh = (readBuff[3] << 8) | readBuff[4];     // Humi
+
+            bool tCrc = CRC8(readBuff.Slice(0, 2), readBuff[2]);
+            bool rhCrc= CRC8(readBuff.Slice(3, 2), readBuff[5]);
+            if (tCrc == false || rhCrc == false)
+            {
+                return;
+            }
+
+            // Details in the Datasheet P13
+            _temperature = Math.Round(st * 175 / 65535.0 - 45, 1);
+            _humidity = Math.Round(srh * 100 / 65535.0, 1);
+        }
+
+        /// <summary>
+        /// 8-bit CRC Checksum Calculation
+        /// </summary>
+        /// <param name="data">Raw Data</param>
+        /// <param name="crc8">Raw CRC8</param>
+        /// <returns>Checksum is true or false</returns>
+        private bool CRC8(ReadOnlySpan<byte> data, byte crc8)
+        {
+            // Details in the Datasheet P13
+            byte crc = CRC_INIT;
+            for (int i = 0; i < 2; i++)
+            {
+                crc ^= data[i];
+
+                for (int j = 8; j > 0; j--)
```

Comment:
why do you go backwards instead of forward? I'd let JIT/compiler take care of any optimizations like that if that was the motivation